src/codegen/object.rs

```diff
@@ -34,12 +34,12 @@ pub fn generate(
     if need_generate_inherent(analysis) {
         try!(writeln!(w));
         try!(write!(w, "impl {} {{", analysis.name));
-        for func_analysis in &analysis.constructors() {
+        for func_analysis in &mut analysis.constructors() {
             try!(function::generate(w, env, func_analysis, false, false, 1));
```

Comment:
I not tried to build this but I don't see change in `function::generate` declaration: is &mut really needed?

Reply:
No it's not, you're absolutely right.